### PR TITLE
Allowing config file to be located on a URL

### PR DESCRIPTION
**Pull request by jsanemet**
Adding the chance of downloading a configuration file from an URL. The downloaded file will be stored on the user's default path: ~/.config/cibyl.yaml. If the file already exists, the user is prompted if they want to override it. If they do not, the application exists with an error. 

I have also changed how the config instances are generated, now a factory will implement its constructor for any type of source, including: path to a file, a URL, search on default paths and app's arguments.

### CODE REVIEW

**Comment by adrianfusco** [comment]
Good job Jorge. In this case should we have a different config? I'm using the same config file with this PR and I receive an error:

```
            system_type: jenkins
[2022-03-31 11:47:12] {afuscoar@afuscoar} (~/.RedHat/Projects/github/cibyl-bak/cibyl) (download-config)$ -> ci --help
Traceback (most recent call last):
  File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/orchestrator.py", line 64, in create_ci_environments
    self.config.data.get('environments', {}).items():
AttributeError: 'NoneType' object has no attribute 'get'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/afuscoar/.local/bin/ci", line 33, in <module>
    sys.exit(load_entry_point('Cibyl==0.0.2', 'console_scripts', 'ci')())
  File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/cli/main.py", line 69, in main
    orchestrator.create_ci_environments()
  File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/orchestrator.py", line 76, in create_ci_environments
    raise InvalidConfiguration from exception
cibyl.exceptions.config.InvalidConfiguration: 
Invalid Configuration.
A valid configuration should specify an environment, its system(s) and the
system(s) details

environments:
    env_1:
        jenkins_system:
            system_type: jenkins
```

**Comment by adrianfusco** [comment]
> > Good job Jorge. In this case should we have a different config? I'm using the same config file with this PR and I receive an error:
> > ```
> >             system_type: jenkins
> > [2022-03-31 11:47:12] {afuscoar@afuscoar} (~/.RedHat/Projects/github/cibyl-bak/cibyl) (download-config)$ -> ci --help
> > Traceback (most recent call last):
> >   File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/orchestrator.py", line 64, in create_ci_environments
> >     self.config.data.get('environments', {}).items():
> > AttributeError: 'NoneType' object has no attribute 'get'
> > 
> > The above exception was the direct cause of the following exception:
> > 
> > Traceback (most recent call last):
> >   File "/home/afuscoar/.local/bin/ci", line 33, in <module>
> >     sys.exit(load_entry_point('Cibyl==0.0.2', 'console_scripts', 'ci')())
> >   File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/cli/main.py", line 69, in main
> >     orchestrator.create_ci_environments()
> >   File "/home/afuscoar/.local/lib/python3.9/site-packages/cibyl/orchestrator.py", line 76, in create_ci_environments
> >     raise InvalidConfiguration from exception
> > cibyl.exceptions.config.InvalidConfiguration: 
> > Invalid Configuration.
> > A valid configuration should specify an environment, its system(s) and the
> > system(s) details
> > 
> > environments:
> >     env_1:
> >         jenkins_system:
> >             system_type: jenkins
> > ```
> 
> It should be the same, it is fails then I broke something. Please pass the configuration file so that I can try it.

Thx. Sent to you.

File: cibyl/exceptions/__init__.py [hunk 1]
```diff
@@ -17,15 +17,21 @@
 
 
 class CibylException(Exception):
+    def __init__(self, message):
```
Comment (cescgina) [inline cibyl/exceptions/__init__.py:20]: I have two questions about this:
1. Do we really need the `__init__`?
2. Why not `super().__init__(message)`?

**Review by cescgina — COMMENTED** [review]
I'm not sure about saving the downloaded at the default path and asking for user input. I would suggest maybe to by default use a temporary file for the downloaded config and add an option to save it in the default path.

File: cibyl/exceptions/__init__.py (hunk 1 above)
Comment (jsanemet) [reply cibyl/exceptions/__init__.py:20]: 1.- Not entirely, but Exception's default constructor is: (*args, **kwargs) and I got a warning from Pylint for it.
2.- Same reason, Pylint warned that *args did not match a str.

Comment (cescgina) [reply cibyl/exceptions/__init__.py]: ok, makes sense. I would pass it as kwarg, still. IMO it's clearer

**Review by cescgina — APPROVED** [review]
Code seems robust enough so that if later we want to change the behaviour it won't be too dificult. Good job!

**Pull request merged**